src/Builder/Visitor/AddPluginToPluginCollectionExtendContainerVisitor.php

```diff
@@ -134,9 +167,59 @@ protected function handleContainerExtendClosure(Closure $closure, array $addPlug
             $newMethodCall = (new BuilderFactory())
                 ->methodCall($addPluginCalls[0]->var, $addPluginCalls[0]->name, $arguments);
 
-            array_splice($closure->stmts, $newPluginAddCallIndex, 0, [new Expression($newMethodCall)]);
+            array_splice($closure->stmts, (int)$newPluginAddCallIndex, 0, [new Expression($newMethodCall)]);
         }
 
         return $closure;
     }
+
+    /**
+     * @param \PhpParser\Node $node
+     *
+     * @return array<string>
+     */
+    protected function getPluginList(Node $node): array
+    {
+        $plugins = [];
+
+        foreach ($node->stmts as $stmt) {
+            if (
+                $stmt instanceof Expression
+                && $stmt->expr instanceof MethodCall
+                && count($stmt->expr->args) >= 2
+                && $stmt->expr->args[1]->value instanceof Closure
+            ) {
+                /** @var \PhpParser\Node\Expr\Closure $closure */
+                $closure = $stmt->expr->args[1]->value;
+                foreach ($closure->stmts as $stmt) {
+                    if ($stmt instanceof Expression === false) {
+                        continue;
+                    }
+                    if (
+                        $stmt->expr instanceof MethodCall === false
+                        || strpos(strtolower($stmt->expr->name->toString()), 'add') === false
+                    ) {
+                        continue;
+                    }
+
+                    /** @var \PhpParser\Node\Arg $arg */
+                    foreach ($stmt->expr->args as $arg) {
+                        if ($arg->value instanceof New_) {
+                            $plugins[] = $arg->value->class->toString();
+                        }
+                    }
+                }
+            }
+        }
+
+        return $plugins;
+    }
+
+    /**
+     * @return \SprykerSdk\Integrator\Builder\Visitor\PluginPositionResolver\PluginPositionResolverInterface
+     */
+    protected function getPluginPositionResolver(): PluginPositionResolverInterface
+    {
+        return new PluginPositionResolver();
+    }
```

Comment:
This doesn't look good for me. I think it's better to extend the constructor with $pluginPositionResolver and extend also all classes that create the object of `AddPluginToPluginCollectionExtendContainerVisitor` and at one point you will come to Factory the creates the object of the class and I think only in factory it is possible to have:

```
function getPluginPositionResolver(): PluginPositionResolverInterface
    {
        return new PluginPositionResolver();
    }
```

On my opinion all other classes except of Factory shouldn't create the objects of the classes by themselves. We should use DI for that.

Reply:
I agree with you. Moved the method to Factory